### PR TITLE
Adding profile pages

### LINKED ISSUE #69: [US-16] As a user I want to have a User Profile page so that I can see my posts and also my "Followers" and who I "Follow" [9]
As a user, I shall be able to see what I have posted before and be able to see a list of whom I am following and who is following me.  

### **Functional Requirements**

- [x] Another page for the profile 
- [x] The profile page should only include the post that the user has posted 
- [x] The profile page should display a list of who is following and who is following the user. 

### **Specifications**

- [x] Retrieve user post in the database within 2 seconds when refreshing the page 
- [x] Retrieve who is following the user from the database within 3 seconds when refreshing the page
- [x] Retrieve Who the user is following from the database within 3 seconds when refreshing the page 

### **Acceptance Criteria**

- [x] Clicking on your username will redirect you to your profile page
- [x] Profile page will pull all posts made by the user from the database and display it on the webpage
- [x] Have a separate webpage to display information regarding "Followers"
- [x] This webpage should be split into 2 halves, one displaying "Followers" and the other showing "Following"
- [x] "Followers" would display all other users that are following the logged in user 
- [x] "Following" would display all the users that the logged in user is following.

### PR DESCRIPTION
**Pull request by maherysm**
#69 
- Each user has a profile page
- Profiles can be visited via the home page by clicking on the posts usernames
- Profile pages show
  - user details (name, number of followings and followers)
  - user posts
  - follow button (unless the profile belongs to the authenticated user)
  - list of followings and followers when clicking respectively on the Followings and Followers

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -142,7 +148,7 @@
     <property name="RunOnceActivity.OpenProjectViewOnStart" value="true" />
     <property name="RunOnceActivity.ShowReadmeOnStart" value="true" />
     <property name="WebServerToolWindowFactoryState" value="false" />
-    <property name="last_opened_file_path" value="$PROJECT_DIR$/../react-laravel-master" />
```
Comment (halloweentown) [inline .idea/workspace.xml:145]: Can we confirm that ".idea/workspace.xml " should be in the project?
I have a feeling it should be in the ignore file as it contains the paths.

File: idk/composer.lock [hunk 2]
```diff
@@ -8,16 +8,16 @@
     "packages": [
```
Comment (halloweentown) [inline idk/composer.lock:8]: Updates to all packages.

**Review by j-loy — APPROVED** [review]
Group project files together with final feature implemented

File: .idea/workspace.xml (hunk 1 above)
Comment (halloweentown) [reply .idea/workspace.xml:145]: **From the link below, we can ignore it only once it is removed:
https://stackoverflow.com/questions/57091527/what-exactly-is-idea-workspace-xml**

**Review by halloweentown — APPROVED** [review]
The following path should be added to gitgnore file:
/idk/public/uploads/


The following paths CAN be added, but we may lose all data in the process.
Proceed with caution:
.idea/workspace.xml
 idk/storage/logs/laravel.log

**Review by halloweentown — APPROVED** [review]

**Comment by halloweentown** [comment]
image
@MahHairy  can you please try the command line resolution proposed by git?

```
git fetch origin
git checkout -b backend-MahHairy-Profile origin/backend-MahHairy-Profile
git merge main
```
```
git checkout main
git merge --no-ff backend-MahHairy-Profile
git push origin main
```

**Comment by halloweentown** [comment]
@giroux2000 @MahHairy and I have not been able to resolve the conflicts cited above using the following methods:
- Github web tool
- GitBash
- Github Desktop

As such, this PR will be closed, and a fresh branch and pull request we're created instead: #84

**Pull request closed without merging**